packages/loader/container-loader/src/container.ts

```diff
@@ -841,8 +842,26 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this._resolvedUrl = resolvedUrl;
             await this.connectStorageService();
 
-            // This we can probably just pass the storage service to the blob manager - although ideally
-            // there just isn't a blob manager
+            // upload blobs here (NYI)
+
+            // post summary here
+            if (this.loader.services.detachedBlobStorage && this.loader.services.detachedBlobStorage.size > 0) {
+                if (this.cachedAttachSummary === undefined) {
+                    const appSummary: ISummaryTree = this.context.createSummary();
+                    const protocolSummary = this.captureProtocolSummary();
+                    this.cachedAttachSummary = combineAppAndProtocolSummary(appSummary, protocolSummary);
+                }
+                const summary = this.cachedAttachSummary;
+
+                await this.storageService.uploadSummaryWithContext(summary, {
+                    referenceSequenceNumber: 0,
+                    ackHandle: undefined,
+                    proposalHandle: undefined,
+                });
+            }
+
+            assert(!this.loader.services.detachedBlobStorage || this.loader.services.detachedBlobStorage.size === 0,
```

Comment:
you might move this assert into above if and make it unconditional - a bit easier to follow logic